healpy/fitsfunc.py

```diff
@@ -721,6 +721,8 @@ def getformat(t):
         np.dtype(np.complex64): "C",
         np.dtype(np.complex128): "M",
     }
+    if hasattr(t, "type"):
+        t = t.type
```

Comment:
@lpsinger this doesn't come out in the unit tests.
But if I run the tutorial notebook, I have:

```
wmap_map_I.dtype
dtype('>f4')
```

If I try to write it to disk:

```
> hp.write_map("my_map.fits", wmap_map_I, overwrite=True)
/extrahome/software/healpy/healpy/fitsfunc.py:199: UserWarning: setting the output map dtype to [dtype('>f4')]
  warnings.warn("setting the output map dtype to %s" % str(dtype))

---------------------------------------------------------------------------
VerifyError                               Traceback (most recent call last)
<ipython-input-15-38a2a4f140b4> in <module>
----> 1 hp.write_map("my_map.fits", wmap_map_I, overwrite=True)

/extrahome/software/healpy/healpy/fitsfunc.py in write_map(filename, m, nest, dtype, fits_IDL, coord, partial, column_names, column_units, extra_header, overwrite)
    241             mm2 = np.asarray(mm)
    242             cols.append(
--> 243                 pf.Column(
    244                     name=cn,
    245                     format="1024%s" % curr_fitsformat,

~/miniconda3/envs/healpyclean/lib/python3.8/site-packages/astropy/io/fits/column.py in __init__(self, name, format, unit, null, bscale, bzero, disp, start, dim, array, ascii, coord_type, coord_unit, coord_ref_point, coord_ref_value, coord_inc, time_ref_pos)
    619                 msg.append(indent(val[1]))
    620 
--> 621             raise VerifyError('\n'.join(msg))
    622 
    623         for attr in KEYWORD_ATTRIBUTES:

VerifyError: The following keyword arguments to Column were invalid:
    Column format option (TFORMn) failed verification: Illegal format `1024None`. The invalid value will be ignored for the purpose of formatting the data in this column.
```

I am adding a unit test about this.

Reply:
It seems like it is caused by endianness, I don't understand why, if the input array is `>f4`, this patch is needed, if it is `f4`, it is not necessary. See test https://github.com/healpy/healpy/pull/688/commits/540ef2ddf90eab20f6c58f70fddf00f6604bbeb3